### DISCUSSION TITLE
Next steps

### DISCUSSION THREAD

**Discussion by fmartingr** [go-shiori/shiori#327] [Announcements]
So here we are! I’ve been taking a quick look at everything and I’m going to try and explain a bit of the changes before anything else.

# Use of the discussions section

First of all I will be getting into “triaging” the already present issues and PRs (some of those date 2018, four years ago!), and that’s one of the changes I want to introduce into the repository: the discussions section.

It’s very typical to have an overflowing issues section on a repository, where people just discuss stuff, ask of help or request enhancements into the repo. In some cases these don’t translate into a specific action over the project, and even if it does, it’s not inmediate.

So (as you can already see) the discussions section will be split in three categories:

- **Announcements:** Where the team will post updates and important things for the community
- **Ideas:** Any request or proposal for the repository that doesn’t have anyone assigned, work is not clear, etc. goes here.
- **Q&A:** For questions related to the installation, configuration and/or usage of Shiori itself. (Maybe I will rename it to "Ask for help here" or something like that).

The most critical here is _Ideas_ and where the line is between having a discussion and an issue about a certain subject. I would say that... if it’s well defined and someone is going to work on it it should have an issue and if not, it should be a discussion. (Im pretty sure we will improve this definition along the way)

**But why?** One of the most demotivating things I see everyday in any forge is having an overflowing issues section. It gives **a lot** of work to maintainers since it needs to be checked, followed on and updated over time, so I think moving the “undefined” tasks into the discussions section will alleviate that, leaving the issues for real pending work on the project.

Also it seems that discussions have the same labels so information won’t be lost, we could tag request with “good first issue” and all that.

# Quality of life for maintainers

I’m not super familiar with the project itself but I’ve seen some improvements for maintainers that would easily translate into a more streamlined experience for new contributors.

There are a few things that require checking:

- Docker images: Are integrations still building? A lot changed in Docker Hub since the last release.
    - Should we make a docker image at all if we provide alternatives?
- Release: How is it done?
    - Having goreleaser here would be awesome.
    - Should we provide linux packages along the binaries/source?
- Documentation:
    - Should it reside on the wiki or under a `docs` folder?
    - Should the API documentation be migrated to Swagger?

And for newcomers/contributors:

- Of course, more communication with the community.
- Providing a `Makefile` full of useful starter commands is always amazing and a good entrypoint for people to look how everything works together
- A contributing documentation with basic guidelines seems necessary as well (what about someone not very familiar with VCSs that want to make a simple change to a template?)

# Closing

I have taken the liberty to create a #shiori IRC channel for more synchronous communication in [libera.chat](https://web.libera.chat/#shiori). Don’t expect much but it always a nice touch to bring a community together.

Please don’t think that we’re full steam ahead and there is going to be a lot of movement all of a sudden. I’m just one person with a life and a full time job that is going to take this **slowly** to prevent burnout.

There’s tons to do! Expect to start seeing me moving things around and commenting on your issues/PRs over the following days, this will take time but we will move it forward together.

Thanks everyone.

**Comment by symgryph** [comment]
We we want to write a code of conduct? I could draft something. I think it will help maintain a better community image if we have clear expectations.

**Reply by fmartingr** [reply]
I believe the [current COC](https://github.com/go-shiori/shiori/blob/master/CODE_OF_CONDUCT.md) is enough for now. It can be reviewed in the future if necessary.

**Comment by wuvei** [comment]
Hi, as for communication, a [discord channel](https://discord.com/) may be a more popular way. I'd recommend creating a discord server for shiori community.

**Reply by fmartingr** [reply]
Personally speaking, I'm not very fond of Discord for software communities, it tends to black hole information and filled with lurkers that do not participate. (How many discord servers on your sidebar that you haven't said a word?). 

Also I don't want to have ownership of anything Shiori related outside of Github, since I'm not the only maintainer. I opened the chat in case there was some wandering shiori users around, but didn't expect it to gain any traction. IRC may be old but is good for ephemeral conversations, anything else can discussed here in github under the discussions tab. 

Of course, this may change in the future if there's a need for such a tool.
